src/integration/resources/inprocess/cluster.go

```diff
@@ -114,33 +133,37 @@ func NewClusterConfigsFromYAML(dbnodeYaml string, coordYaml string, aggYaml stri
 
 // NewCluster creates a new M3 cluster based on the ClusterOptions provided.
 // Expects at least a coordinator, a dbnode and an aggregator config.
-func NewCluster(configs ClusterConfigs, opts resources.ClusterOptions) (resources.M3Resources, error) {
-	if err := opts.Validate(); err != nil {
+func NewCluster(
+	configs ClusterConfigs,
+	opts resources.ClusterOptions,
+) (resources.M3Resources, error) {
+	fullConfigs, err := GenerateClusterFullConfigs(configs, opts)
+	if err != nil {
 		return nil, err
 	}
 
-	logger, err := resources.NewLogger()
-	if err != nil {
+	return NewClusterFromFullConfigs(fullConfigs, opts)
+}
+
+// NewClusterFromFullConfigs creates a new M3 cluster with the given full
+// set of configuration and options.
```

Comment:
Might be nice to outline the difference between ClusterConfigs and ClusterFullConfigs. I understand the cluster full configs contain the config for each instance of a component, but what else is different and when would I use one vs the other as a client of the test framework?